### ISSUE TITLE
Certain pattern fails to match with JIT complete flag

### ISSUE THREAD

**Issue by TETYYS** [ltrzesniewski/pcre-net#22]
Following code returns false:
```
var regex = new PcreRegex("[\\w]*[CA]X*B", PcreOptions.Compiled);

var match = regex.Match("ABC");
return match.Success;
```

While following returns `0: AB`:
```
#include <stdlib.h>

#define PCRE2_CODE_UNIT_WIDTH 8
#include <pcre2.h>

int main()
{
    int errornumber, rc;
    PCRE2_SIZE erroroffset;

    pcre2_code *rgx;
    pcre2_match_context *mCtx;
    pcre2_compile_context* context;
    pcre2_match_data *matchData;
    pcre2_jit_stack *jitStack;
    int ret;

    mCtx = pcre2_match_context_create(NULL);
    context = pcre2_compile_context_create(NULL);

    const char *pattern = "[\\w]*[CA]X*B";

    rgx = pcre2_compile(pattern, strlen(pattern), 0, &errornumber, &erroroffset, context);
	if (rgx == NULL) {
        char buffer[256];
        int messageLength = pcre2_get_error_message(errornumber, buffer, 256);

        printf("%.*s, %d\n", messageLength, buffer, erroroffset);
		return 1;
	}

    rc = pcre2_jit_compile(rgx, PCRE2_JIT_COMPLETE);
    if (rc == PCRE2_ERROR_JIT_BADOPTION) {
        printf("no jit\n");
    }

    jitStack = pcre2_jit_stack_create(32*1024, 512*1024, NULL);
    pcre2_jit_stack_assign(mCtx, NULL, jitStack);

    matchData = pcre2_match_data_create_from_pattern(rgx, NULL);

    ret = pcre2_jit_match(rgx, "ABC", 3, 0, 0, matchData, mCtx);

    if (ret < 1) {
        printf("%d\n", ret);
    } else {
        PCRE2_SIZE *ovector = pcre2_get_ovector_pointer(matchData);

        for (int i = 0; i < ret; i++)
        {
            PCRE2_SPTR substring_start = "ABC" + ovector[2*i];
            size_t substring_length = ovector[2*i+1] - ovector[2*i];
            printf("%2d: %.*s\n", i, (int)substring_length, (char *)substring_start);
        }
    }
}
```
I tried both `pcre2_match` and `pcre2_jit_match`. Could be that the issue is somewhere in usage of UTF16 APIs? Maybe return value of jit compile says something here https://github.com/ltrzesniewski/pcre-net/blob/master/src/PCRE.NET.Native/pcrenet_compile.c#L66

**Comment by TETYYS** [comment]
This succeds, though:
```
var regex = new PcreRegex("[\\w]*[CA]X*B", PcreOptions.CompiledPartial);

var match = regex.Match("ABC");
Assert.True(match.Success);
Assert.False(match.IsPartialMatch);
```

**Comment by ltrzesniewski** [comment]
Thanks, I can reproduce it. I'll look into it.

**Comment by ltrzesniewski** [comment]
It looks like an issue in PCRE2. I ran the following with v10.35:

```
$ ./configure --enable-jit
$ make
```

And then:

```
$ echo ABC | ./pcre2grep --no-jit '[\w]*[CA]X*B'
ABC

$ echo ABC | ./pcre2grep '[\w]*[CA]X*B'
```

I also noticed that using `PCRE2_NO_START_OPTIMIZE` bypasses the issue:

```
$ echo ABC | ./pcre2grep '(*NO_START_OPT)[\w]*[CA]X*B'
ABC
```

Note that I tested this with the default code unit width of 8. I'm not sure why your C version works though. Have you compiled it against PCRE2 v10.35 or a different version?

Anyway, I suggest you should report this issue in the [PCRE bug tracker](https://bugs.exim.org/enter_bug.cgi?product=PCRE).

**Comment by TETYYS** [comment]
I compiled it against 10.32, maybe my C code is just badly implemented

**Comment by TETYYS** [comment]
Seems like bug introduced somewhere between 10.32 and 10.35. I get `-1` for 10.35 and `0: AB` for 10.32 with the C program.

**Comment by TETYYS** [comment]
It's fixed in latest TRUNK, so PCRE.NET needs to update when PCRE has a new version

**Comment by ltrzesniewski** [comment]
This is fixed in v0.14.0 (PCRE2 v10.36)
